services/app-web/src/s3/s3Amplify.ts

```diff
@@ -62,22 +64,38 @@ function newAmplifyS3Client(bucketName: string): S3ClientT {
             }
         },
 
-        deleteFile: async (filename: string): Promise<void | S3Error> => {
-            try {
-                await Storage.remove(filename)
-                return
-            } catch (err) {
-                assertIsS3PutError(err)
-                if (err.name === 'Error' && err.message === 'Network Error') {
-                    console.log('Error deleting file', err)
-                    return {
-                        code: 'NETWORK_ERROR',
-                        message: 'Error deleting file from the cloud.',
+        deleteFile: async (
+            filename: string,
+            planPackageStatus?: HealthPlanPackageStatusType,
+            fileItems?: FileItemT[]
+        ): Promise<void | S3Error> => {
+            // Only delete file from S3 bucket if submission status is DRAFT or if file has NOT been submitted before.
+            // Meaning it was uploaded, but health plan package has not been submitted/resubmitted yet.
+            const isSubmittedFile =
```

Comment:
So, I don't think this solves our problem exactly. We need to be checking if a file has _ever_ been uploaded for this HealthPlanPackage, not just the most recent revision.